tests/Doctrine/Tests/Mocks/ConnectionMock.php

```diff
@@ -74,7 +78,17 @@ public function insert($tableName, array $data, array $types = [])
      */
     public function executeUpdate($query, array $params = [], array $types = [])
     {
-        $this->_executeUpdates[] = ['query' => $query, 'params' => $params, 'types' => $types];
+        throw new BadMethodCallException(sprintf('Call to deprecated method %s().', __METHOD__));
+    }
```

Comment:
Was this something you introduced to spot the deprecated calls and if yes, should we remove it? Or is it in order to prevent future contributors from adding more calls?

Reply:
Yes, no and yes. 🙂 

This is a mock class that is used in unit tests only. I think it‘s helpful for future contributions if the mock throws on method calls that we want to avoid.